src/borg/helpers.py

```diff
@@ -142,17 +142,29 @@ def __delitem__(self, name):
         name = safe_encode(name)
         del self._archives[name]
 
-    def list(self, sort_by=None, reverse=False, prefix=''):
+    def list(self, sort_by=(), reverse=False, prefix='', first=None, last=None):
         """
         Inexpensive Archive.list_archives replacement if we just need .name, .id, .ts
-        Returns list of borg.helpers.ArchiveInfo instances
+        Returns list of borg.helpers.ArchiveInfo instances.
+        sort_by can be a list of sort keys, they are applied in reverse order.
         """
+        if isinstance(sort_by, (str, bytes)):
+            raise TypeError('sort_by must be a sequence of str')
         archives = [x for x in self.values() if x.name.startswith(prefix)]
-        if sort_by is not None:
-            archives = sorted(archives, key=attrgetter(sort_by))
-        if reverse:
+        for sortkey in reversed(sort_by):
+            archives.sort(key=attrgetter(sortkey))
+        if reverse or last:
             archives.reverse()
-        return archives
+        n = first or last or len(archives)
+        return archives[:n]
+
+    def list_considering(self, args):
```

Comment:
At first wanted to suggest a different name, but this is actually quite nice... `archives.list_considering(args)`